src/IndexMarkerV2.sol

```diff
@@ -0,0 +1,298 @@
+import "openzeppelin-upgradeable/proxy/utils/Initializable.sol";
+import "openzeppelin-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+import "openzeppelin-upgradeable/token/common/ERC2981Upgradeable.sol";
+import "openzeppelin-upgradeable/token/ERC721/ERC721Upgradeable.sol";
+import "openzeppelin-upgradeable/token/ERC721/extensions/ERC721VotesUpgradeable.sol";
+import "openzeppelin-upgradeable/token/ERC721/extensions/IERC721MetadataUpgradeable.sol";
+import "openzeppelin-upgradeable/utils/cryptography/EIP712Upgradeable.sol";
+import "zora-drops-contracts/interfaces/IOperatorFilterRegistry.sol";
+import "./IRecoveryChildV1.sol";
+import "./utils/IERC173.sol";
+
+/**
+ * note: This contract uses role-based access control. it also has an IERC173
+ * compliant Owner, but this is solely for opensea royalty compatibility. The
+ * Owner is the openseaRoyaltiesManager, which is set by whoever has the
+ * OPENSEA_ROYALTIES_ADMIN_ROLE. The openseaRoyaltiesManager must be a
+ * trusted EOA (not a contract), but the OPENSEA_ROYALTIES_ADMIN_ROLE should
+ * be a multisig or governance contract.
+ */
+
+contract IndexMarkerV2 is
+  Initializable,
+  ERC721Upgradeable,
+  AccessControlUpgradeable,
+  EIP712Upgradeable,
+  ERC721VotesUpgradeable,
+  ERC2981Upgradeable,
+  UUPSUpgradeable,
+  IERC173 /* required for opensea royalties */
+{
+  bytes32 public constant UPGRADER_ROLE = keccak256("UPGRADER_ROLE");
+  bytes32 public constant SET_TOMB_VOTING_WEIGHT_ROLE =
+    keccak256("SET_TOMB_VOTING_WEIGHT_ROLE");
+  bytes32 public constant SET_TOMB_CONTRACTS_ROLE =
+    keccak256("SET_TOMB_CONTRACTS_ROLE");
+  bytes32 public constant SET_ROYALTIES_ROLE = keccak256("SET_ROYALTIES_ROLE");
+  bytes32 public constant UPDATE_MARKET_SETTINGS_ROLE =
+    keccak256("UPDATE_MARKET_SETTINGS_ROLE");
+  bytes32 public constant OPENSEA_ROYALTIES_ADMIN_ROLE =
+    keccak256("OPENSEA_ROYALTIES_ADMIN_ROLE");
+
+  uint16 internal constant DEFAULT_TOMB_HOLDER_VOTING_WEIGHT = 30; // 30 votes
+  uint96 internal constant DEFAULT_ROYALTY_BPS = 1_000; // 10%
+  address payable internal constant TOMB_ARTIST =
+    payable(0x4a61d76ea05A758c1db9C9b5a5ad22f445A38C46);
+
+  // zora & opensea compatibility
+  IOperatorFilterRegistry immutable operatorFilterRegistry =
+    IOperatorFilterRegistry(0x000000000000AAeB6D7670E522A718067333cd4E);
+  address public marketFilterDAOAddress;
+
+  uint96 public royaltyBps;
+  IERC721MetadataUpgradeable public indexMarker;
+  mapping(address => bool) public isTombContract;
+  mapping(address => mapping(uint256 => bool)) public isSingletonTombToken;
+  uint16 public tombHolderVotingWeight;
+  address internal openseaRoyaltiesManager;
+
+  constructor() {
+    _disableInitializers();
+  }
+
+  function initialize(
+    address _indexMarker,
+    address payable _royaltyDestination,
+    address _marketFilterDAOAddress
+  ) public initializer {
+    __ERC721_init("Wrapped Tomb Index Marker", "WMKR");
```

Comment:
let's go with `"Tomb Index Marker"` and `"MKR"`

Reply:
cool, i meant to change it to "Tomb Index Marker V2" and "MKR2". let me know which you prefer between this and what you just said ^

Reply:
i went ahead with "Tomb Index Marker" and "MKR" but lmk if you want me to append " V2" and "2".